### PR TITLE
Splitting base

### PR DESCRIPTION
**Pull request by mmatera**
In this PR, `Centralities`,  `_Collections` and `GraphBox` have their own submodules.

### CODE REVIEW

**Comment by rocky** [comment]
Thanks for doing this!

**Comment by mmatera** [comment]
@rocky, can you remember me the way in which a module can be marked to do not produce documentation?

**Comment by rocky** [comment]
> @rocky, can you remember me the way in which a module can be marked to do not produce documentation?

```python
no_doc = True
````

https://github.com/Mathics3/mathics-core/blob/master/mathics/builtin/base.py#L55-L56

**Comment by rocky** [comment]
LGTM

File: pymathics/graph/base.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from typing import Callable, Optional, Tuple
```
Comment (rocky) [inline pymathics/graph/base.py:14]: Tuple is no longer used. Nor is `base64` or `tempfile` on lines above.  Running this over flake8 will find other things.

File: pymathics/graph/centralities.py [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+class BetweennessCentrality(_Centrality):
+    """
+    <url>
+    :Betweenness centrality:
+    https://en.wikipedia.org/wiki/Betweenness_centrality</url> (<url>
+    :NetworkX:
+    https://networkx.org/documentation/networkx-2.8.8/reference/algorithms/generated/\
+    networkx.algorithms.centrality.betweenness_centrality.html</url>,
+    <url>
```
Comment (rocky) [inline pymathics/graph/centralities.py:76]: Line split should have `\` or this added to above line.

File: pymathics/graph/centralities.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+      <dd>gives a list of Katz centralities for the \
+          vertices in the graph $g$ and weight $alpha$.
+      <dt>'KatzCentrality'[$g$, $alpha$, $beta$]
+      <dd>gives a list of Katz centralities for the \
+          vertices in the graph $g$ and weight $alpha$ and initial centralities $beta$.
+    </dl>
+
+    >> g = Graph[{a -> b, b -> c, c -> d, d -> e, e -> c, e -> a}]
+     = -Graph-
+    >> g
```
Comment (rocky) [inline pymathics/graph/centralities.py:344]: This example should be removed.

File: pymathics/graph/centralities.py [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+    <dl>
+      <dt>'PageRankCentrality'[$g$, $alpha$]
+      <dd>gives a list of page rank centralities for the \
+          vertices in the graph $g$ and weight $alpha$.
+      <dt>'PageRankCentrality'[$g$, $alpha$, $beta$]
+      <dd>gives a list of page rank centralities for the \
+          vertices in the graph $g$ and weight $alpha$ and initial centralities $beta$.
+    </dl>
+
+    # Not working, possibly because an issue in networkx
```
Comment (rocky) [inline pymathics/graph/centralities.py:404]: Comment is `##` not `#`

**Pull request merged**